src/Servers/IIS/IIS/test/Common.FunctionalTests/WindowsAuthTests.cs

```diff
@@ -58,4 +60,51 @@ public async Task WindowsAuthTest(TestVariant variant)
         Assert.StartsWith("Windows:", responseText);
         Assert.Contains(Environment.UserName, responseText);
     }
+
+    [ConditionalTheory]
+    [RequiresIIS(IISCapability.WindowsAuthentication)]
+    [MemberData(nameof(TestVariants))]
+    public async Task WindowsAuthWithImpersonationLevelTest(TestVariant variant)
+    {
+        var deploymentParameters = Fixture.GetBaseDeploymentParameters(variant);
+        deploymentParameters.SetAnonymousAuth(enabled: false);
+        deploymentParameters.SetWindowsAuth();
+
+        // The default in hosting sets windows auth to true.
+        var deploymentResult = await DeployAsync(deploymentParameters);
+
+        var impersonationLevels = new TokenImpersonationLevel[]
+            {
+                TokenImpersonationLevel.None,
+                TokenImpersonationLevel.Identification,
+                TokenImpersonationLevel.Impersonation,
+                TokenImpersonationLevel.Delegation,
+                TokenImpersonationLevel.Anonymous
+            };
+
+        foreach (var impersonationLevel in impersonationLevels)
+        {
+            // TokenImpersonationLevel is not supported by HttpClient so we need to use HttpWebRequest to test it.
+#pragma warning disable SYSLIB0014 // Type or member is obsolete
+            var request = HttpWebRequest.CreateHttp($"{deploymentResult.HttpClient.BaseAddress}Auth");
+#pragma warning restore SYSLIB0014 // Type or member is obsolete
+            request.ImpersonationLevel = impersonationLevel;
```

Comment:
I was going to say ImpersonationLevel isn't supported, but went to check the source code in case anything has changed and was surprised to see that they use reflection to enable it for HttpWebRequest, even though HttpClientHandler/SocketsHttpHandler don't expose the api. Time to open an issue as this is an important missing feature that WCF needs.